lib/transport/selenium-webdriver/browserstack.js

```diff
@@ -8,15 +8,32 @@ class Browserstack extends Selenium {
   }
 
   get accessKey() {
-    return this.settings.desiredCapabilities['browserstack.key'];
+    return this.settings.desiredCapabilities['bstack:options'] &&
+      this.settings.desiredCapabilities['bstack:options'].accessKey
+      ? this.settings.desiredCapabilities['bstack:options'].accessKey
+      : this.settings.desiredCapabilities['browserstack.key'];
+
```

Comment:
Only change required over here would be:

```
get accessKey() {
    return this.settings.desiredCapabilities['bstack:options'].accessKey;
  }
```

Reply:
This would be insufficient in case the environment variables haven't been set and creds are passed only in `desiredCapabilities` at root level as `browserstack.key`.. same with `browserstack.user`

Reply:
It would work in that case too

Reply:
How about we make changes in `adaptSettings` to handle this case, something like this:

```
 if (!this.accessKey && this.settings.desiredCapabilities['browserstack.key']) {
      this.settings.desiredCapabilities['bstack:options'].accessKey = this.settings.desiredCapabilities['browserstack.key'];
    }

```

Reply:
The change was not for the purpose of copying over root caps into bstack:options.
To explain, this is current code in adaptSettings:
```
if (!this.accessKey && process.env.BROWSERSTACK_KEY) {
    this.settings.desiredCapabilities['browserstack.key'] = process.env.BROWSERSTACK_KEY;
}
```
and `this.accessKey` returns `this.settings.desiredCapabilities['browserstack.key']`

so adaptSettings is adding the `process.env.BROWSERSTACK_KEY` into caps only when it is not set in `this.settings.desiredCapabilities['browserstack.key']` .. the intended change is to only add the `process.env.BROWSERSTACK_KEY` into `desiredCaps` when both `this.settings.desiredCapabilities['browserstack.key']` **AND** `this.settings.desiredCapabilities['bstack:options'].accessKey` are unset, hence the change in the `get accessKey` function.

Reply:
@Ankit098, It was not for that purpose but we can do that and make the code look cleaner